packages/lib-classifier/src/store/DrawingStore.js

```diff
@@ -1,56 +1,175 @@
-import { addDisposer, getRoot, onAction, types } from 'mobx-state-tree'
+import cuid from 'cuid'
+import { addDisposer, getRoot, getType, isValidReference, onAction, types } from 'mobx-state-tree'
 import { autorun } from 'mobx'
 import { Subject } from 'rxjs'
+import { filter, map, skipUntil } from 'rxjs/operators'
+
+import { LineStore, PointStore } from './markings'
+
+const markStores = {
+  line: LineStore,
+  point: PointStore
+}
+
+function getMarkStore (toolType) {
+  return markStores[toolType] || null
+}
 
 const DrawingStore = types
   .model('DrawingStore', {
-    activeDrawingTool: types.optional(types.number, 0)
+    activeDrawingToolIndex: types.optional(types.number, 0),
+    activeMark: types.safeReference(types.union({
+      dispatcher: (snapshot) => {
+        const snapshotType = getType(snapshot)
```

Comment:
Something I learnt while working on #1212: `getType` can only be called on a tree node, not on a snapshot. Here it will return `LineStore` for a line or `PointStore` for a point so the rest of the function becomes `return snapshotType` which can then be shortened further to `dispatcher: getType`. Again, that only works if you're passing around valid tree nodes, not JSON snapshots of models.